src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,102 @@
+     * @throws DukeException if user input string is not a valid format
+     */
+    public static Command getCommand(String input) throws DukeException {
+        if (input.trim().equals("bye")){
+            return new ExitCommand();
+        } else if (input.trim().equals("list")) {
+            return new ListCommand();
+        } else if (input.matches("done (.*)")) {
+            try {
+                String text = input.substring(5);
+                int taskNo = Integer.parseInt(text);
+                return new DoneCommand(taskNo);
+            } catch (NumberFormatException e) {
+                String str = "Please use the format done <task No.>";
+                throw new DukeException(str);
+            }
+        } else if (input.matches("delete (.*)")) {
+            try {
+                String text = input.substring(7);
+                int taskNo = Integer.parseInt(text);
+                return new DeleteCommand(taskNo);
+            } catch (NumberFormatException e) {
+                String str = "Please use the format delete <task No.>";
+                throw new DukeException(str);
+            }
+
+        } else if (input.matches("todo(.*)")) {
+            try {
+                int start = input.indexOf("todo ");
+                String validInput = input.substring(start);
+                String taskDesc = validInput.trim().substring(5);
+                return new ToDoCommand(taskDesc);
+            } catch(StringIndexOutOfBoundsException e) {
+                throw new DukeException("The todo task description cannot be empty. "
+                        + "Please use format todo <desc>");
+            }
+        } else if (input.matches("deadline(.*)")) {
+            try {
+                int start = input.indexOf("deadline ");
+                String validInput = input.substring(start);
+                String taskDesc = validInput.substring(9);
+                String[] fields = taskDesc.split(" /by ", 2);
+                LocalDate date = LocalDate.parse(fields[1]);
+                return new DeadlineCommand(fields);
+            } catch (DateTimeParseException e) {
+                throw new DukeException("Please use format deadline <description> /by <yyyy-mm-dd>.");
+            } catch (StringIndexOutOfBoundsException e) {
+                throw new DukeException("The Deadline description cannot be empty. "
+                        + "Please use format deadline <description> /by <time>.");
+            } catch (ArrayIndexOutOfBoundsException e) {
+                throw new DukeException("Please use format deadline <description> /by <time>.");
+            }
+        } else if (input.matches("event(.*)")) {
+            try {
+                int start = input.indexOf("event ");
+                String validInput = input.substring(start);
+                String taskDesc = validInput.substring(6);
+                String[] fields = taskDesc.split(" /at ", 2);
+                //make sure not empty
```

Comment:
Maybe rephrase as `Ensure that it is not empty`? Additionally, maybe it could be more descriptive (e.g. what is not empty)?

From [guildeline on comments](https://se-education.org/guides/conventions/java/intermediate.html#comments):
>All comments should be written in English.
>Furthermore, use American spelling and avoid local slang.
 >Rationale: The code is meant for an international audience.